test/unit/sites/greenman-gaming-test.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const GreenmanGamingSite = require('../../../lib/sites/greenman-gaming');
+const cheerio = require('cheerio');
+const siteUtils = require('../../../lib/site-utils');
+
+const VALID_URI = 'http://greenmangaming.com/123/product';
+const INVALID_URI = 'http://www.bad.com/123/product';
+
+describe('The GreenmanGaming Site', () => {
+  it('should exist', () => {
+    expect(GreenmanGamingSite).toBeDefined();
+  });
+
+  describe('isSite() function', () => {
+    it('should return true for a correct site', () => {
+      expect(GreenmanGamingSite.isSite(VALID_URI)).toBeTruthy();
+    });
+
+    it('should return false for a bad site', () => {
+      expect(GreenmanGamingSite.isSite(INVALID_URI)).toBeFalsy();
+    });
+  });
+
+  it('should throw an exception trying to create a new site with an incorrect uri', () => {
+    expect(() => {
+      /* eslint-disable no-new */
+      new GreenmanGamingSite(INVALID_URI);
+      /* eslint-enable no-new */
+    }).toThrow();
+  });
+
+  describe('a new GreenmanGaming Site', () => {
+    let site;
+
+    beforeEach(() => {
+      site = new GreenmanGamingSite(VALID_URI);
+    });
+
+    it('should exist', () => {
+      expect(site).toBeDefined();
+    });
+
+    it('should return the same URI for getURIForPageData()', () => {
+      expect(site.getURIForPageData()).toEqual(VALID_URI);
+    });
+
+    it('should return false for isJSON()', () => {
+      expect(site.isJSON()).toBeFalsy();
+    });
+
+    describe('with a populated page', () => {
+      let $;
+      let bad$;
+      let price;
+      let category;
+      let name;
+
+      beforeEach(() => {
+        price = 4199.3;
+        category = siteUtils.categories.VIDEO_GAMES;
+        name = 'Homefront: The Revolution';
+
+        $ = cheerio.load(`<strong class="curPrice">₹4199.30</strong>
+          <h1 itemprop="name">Homefront: The Revolution</h1>
+          <script>var utag_data = {"currency_code": "INR",
+          "request_path": "/s/in/en/pc/games/action/far-cry-primal/",
+          "product_price_readable": "4199.30"};</script>`);
```

Comment:
For the same reason, I stopped using the .curPrice or any selector! The data was inconsistent and URL was unpredictable! But of all the regions I've checked, the JSON object remains same. So I thought I better go with JSON object to get both price as well as currency. Because it is much more consistent across all regions.